addons/account_payment/models/payment_provider.py

```diff
@@ -24,7 +24,7 @@ def _ensure_payment_method_line(self, allow_create=True):
             return
 
         pay_method_line = self.env['account.payment.method.line'].search(
-            [('code', '=', self.code), ('payment_provider_id', '=', self.id)],
+            [('payment_provider_id', '=', self.id)],
```

Comment:
This is not supposed to change anything. I think the problem is somewhere else. I'm even not sure Wire Transfer is supposed to have a journal since it doesn't create any payment.transaction :thinking: 

What about:
```py
            default_payment_method_id = self._get_default_payment_method_id(self.code)
            if not default_payment_method_id:
                return  # No payment method linked to this acquirer.
```

Reply:
Well, if it is supposed to work, why do I get this? :)
```
In [13]: env['account.payment.method.line'].search([("code", "=", "custom")])
Out[13]: account.payment.method.line()

In [14]: env['account.payment.method.line'].search([("payment_provider_id.code", "=", "custom")])
Out[14]: account.payment.method.line(632,)
```
Funny thing is that it works if I search w/ code=manual. Looks like there's something fishy when the code is "custom".

> if not default_payment_method_id:

This won't work AFAIS because you always have a method since it falls back to `account.account_payment_method_manual_in`. No?

Reply:
The fallback seems useless

Reply:
It's not, it's giving back the default one:
```
In [28]: provider.name
Out[28]: 'Wire Transfer'

In [29]: provider._get_provider_payment_method(provider.code)
Out[29]: account.payment.method()

In [30]: provider._get_default_payment_method_id(provider.code)
Out[30]: 1

In [31]: env.ref('account.account_payment_method_manual_in')
Out[31]: account.payment.method(1,)
```

Reply:
@smetl ping :)

Reply:
@simahawk I'm lost :face_with_spiral_eyes: 
To me, 1 acquirer with API = 1 payment method.
For wire transfer, the whole thing has to be made without any payment.transaction. That's why we don't have any payment method "Wire Transfer". Also, I don't know why we have 2 methods:
_get_provider_payment_method & _get_default_payment_method_id. _get_default_payment_method_id seems removed during the forward-port... is it still used?

Reply:
Well, this piece of code is so contorted that is hard to understand why is there and how it is supposed to work. I understand you being lost :sweat_smile: 

Which FWD port? Those methods are used here in v16.

Now, regarding the WT maybe you are right but you have to consider that the same "custom" logic might apply to custom payment providers that might not need an api and still need to register payments.
That's the whole point about `payment_custom`... 

In our use case, the customer has real transactions happening somewhere else and we only want to collect them and generate their payment lines. Hence, `payment_custom` is the way to go.

Yet, the issue is not about registering transactions. Here it breaks on the auto-configuration. 
Issue is:

1. it's not clear why you should compute the journal every time
2. it's not clear why you should create/delete lines every time it happens

However, very likely it has been done in the past for a reason... unknown to us... but we have to try to make it work. Hence, we need to fix this search of the line which is the last broken thing in the flow.

I would simply remove the code in the 1st search as done and in the second search I'd replace `code` w/ `payment_provider_id.code`. This way it should work w/o altering any implementation.

Reply:
@simahawk But 1) the code has to be correct regarding the payment acquirer and 2) it doesn't make any sense to create a payment method line for Wire Transfer since no payment transaction will never be created :sweat_smile: 
Idk why we end up with so many code on payment acquirer because it's just a matter of knowing if there is a payment transaction API implemented or not at this point... 

> This way it should work w/o altering any implementation.

Since we completely change a lot of things regarding the payment method lines, it's too late. The real question is to know how it's supposed to work.

@AntoineVDV Can you help us on this plz? `custom` means we never need to create any payment.transaction, correct?

Reply:
> @AntoineVDV Can you help us on this plz? `custom` means we never need to create any payment.transaction, correct?

No, it means that there is no payment API to integrate with and that the transaction status tracking is implemented with a custom flow (`provider.custom_mode`). E.g., Wire transfers (no tracking), Pay in Store (manual update), SEPA debits (updated on bank statement sync), etc.

Reply:
So, the whole problem is the search at the beginning of the method that doesn't correspond to the creation of the payment method line. I think we need to
1) align the search with the create by searching on the payment_method instead of the code and create the payment method line using the payment method too.
2) again, regarding the answer from @AntoineVDV , I don't think we are supposed to create "manual" payment method line from a provider. In case of wire transfer, don't return the manual payment method and then, don't create any payment method line.

Since the code of the provider is messy with `code`, `custom_code` and `_get_code`, I think relying on the linked payment method is the safest way to determine the configuration of the accounting part.